app/com/arpnetworking/metrics/portal/alerts/impl/PluggableAlertRepository.java

```diff
@@ -215,7 +219,9 @@ public void addOrUpdateAlert(final Alert alert, final Organization organization)
                             .build();
             mapBuilder.put(uuid, alert);
         }
-        return mapBuilder.build();
+        synchronized (this) {
+            _alerts = mapBuilder.build();
+        }
```

Comment:
Why the `synchronized`? I'd have thought this statement was already atomic:
- `mapBuilder.build()` is pure (and therefore effectively atomic);
- member assignment is atomic (right???);
- therefore `_alerts = mapBuilder.build()` is atomic.

Reply:
AFAIK, yes member assignment is atomic.

I actually had omitted the synchronized initially because I had the same thoughts. I suppose at this point it's moreso a matter of thread visibility rather than safety but even then I would need to make the reads themselves synchronized.

I'm inclined to remove it for now. I'm also left wondering if this could cause issues for the pagination of alerts when we are submitting them for execution (what if it changes mid pagination?). Perhaps we just kick the coordinator?

Reply:
Ahhh. That's a good point. This `synchronized` isn't necessary to stop _other_ code from interfering with _this_ code, but it _does_ stop _this_ code from interfering with _other_ synchronized code.

No strong preferences on this!